tripy/tripy/frontend/tensor.py

```diff
@@ -224,11 +224,15 @@ def eval(self) -> runtime.MemRefValue:
         assert len(data) == 1, "Expects only one output from mlir_tensorrt.compiler executor"
         data = data[0]
 
-        Storage.build_internal([], [self.trace_tensor], data)
+        # If we are tracing, we should not replace the tensor with a storage op. However, even so,
+        # evaluating while tracing could introduce errors (e.g., if the evaluated value is used to create a constant
+        # that is in the graph) and is likely also to slow down compilation, so we still give warnings in that case.
+        if not self.trace_tensor.is_compile_tracer:
+            Storage.build_internal([], [self.trace_tensor], data)
```

Comment:
The problem with this is if we evaluate multiple tensors in a compiled graph, we'll get quadratic time complexity. What I suggested in https://github.com/NVIDIA/TensorRT-Incubator/issues/409#issuecomment-2501952349 is to store the evaluated result but only use it for evaluation and not tracing. I'm not sure exactly how that would work, but I expect it would require changes to how we trace during `eval()`.